### PR TITLE
Replace Zoom registration form by custom one

### PR DESCRIPTION
**Pull request by everthemore**
Endpoint for form submission still to be updated

### CODE REVIEW

**Comment by everthemore** [comment]
TODO: combine mailing list registration in the same form.

**Comment by akhmerov** [comment]
We should still keep a link to the original registration form as a failsafe.

**Comment by everthemore** [comment]
> We should still keep a link to the original registration form as a failsafe.

Turns out this is actually still there from the original, it is part of the jinja template.

File: bots/render_talks.py [hunk 1]
```diff
@@ -7,6 +7,102 @@
...
+<body>
+<form id="registrationForm" method="post" action="vsf-worker.virtualscienceforum.workers.dev/register" style="border:1px solid #ccc">
+  <div class="container">
+    <h1>Sign Up</h1>
+    <p>Please fill in this form to register for the talk by SPEAKERNAME.</p>
+    <hr>
+    <label for="firstname"><b>First Name</b></label>
+    <input type="text" placeholder="Enter your first name" name="firstname" id="name" required>
+    <label for="lastname"><b>Last Name</b></label>
+    <input type="text" placeholder="Enter your last name" name="lastname" id="name" required>
```
Comment (akhmerov) [inline bots/render_talks.py:81]: AFAIU `id`s should be unique in proper HTML.

File: bots/render_talks.py [hunk 2]
```diff
@@ -7,6 +7,102 @@
...
+
+    <label for="address"><b>Email</b></label>
+    <input type="email" placeholder="Enter your Email" name="address" id="address" required>
+    <label for="addressconfirm"><b>Confirm Email</b></label>
+    <input type="email" placeholder="Confirm your Email" name="addressconfirm" id="address" required>
```
Comment (akhmerov) [inline bots/render_talks.py:86]: Do we really need to confirm email?

File: templates/long_range_colloquium.md.j2 [hunk 3]
```diff
@@ -29,7 +29,7 @@ Attendance is open to everyone, but please register to attend! If you would like
 {% if talk.registration_url %}
 If the form below doesn't work, [register directly]({{ talk.registration_url }})
 
-<iframe width="100%" height="940" src="{{ talk.registration_url }}" frameborder="0" marginheight="0" marginwidth="0" style="overflow-x:hidden"></iframe>
+<iframe width="100%" height="940" src="{{ talk.registration_form }}" frameborder="0" marginheight="0" marginwidth="0" style="overflow-x:hidden"></iframe>
```
Comment (akhmerov) [inline templates/long_range_colloquium.md.j2:32]: Will this work? The source of `talk.registration_form` has quotation marks, doesn't this break the HTML?

File: bots/render_talks.py [hunk 4]
```diff
@@ -21,6 +117,11 @@ def format_title(s):
 )
 env.filters['format_title'] = format_title
 
+def update_registration_form(talk):
+    talk['registration_form'] = registration_form_html.replace("SPEAKERNAME", talk['speaker_name']).replace("MEETINGID", talk['zoom_meeting_id'])
+
+env.filter['update_registration_form'] = update_registration_form
```
Comment (akhmerov) [inline bots/render_talks.py:123]: I don't think filters can modify data structures. I'll try something else.

File: bots/render_talks.py (hunk 2 above)
Comment (everthemore) [reply bots/render_talks.py:85]: No, delete?

**Comment by everthemore** [comment]
This is now ready to go, but has one last cosmetic 'issue'. After clicking the new 'register' button, the response comes in the form of a blank page with a success message. To make this a bit nicer, like in the mailinglist signup, we could add another signup function to index.html that handles this. Thoughts?

**Comment by akhmerov** [comment]
Hm, can we repurpose the same form to serve both?

**Pull request merged**